### DISCUSSION TITLE
Blurred model on iOS

### DISCUSSION THREAD

**Discussion by nemesis-ren** [google/model-viewer#3343] [Q&A]
![image](https://user-images.githubusercontent.com/62089287/161769039-284e0289-e3b3-4220-8e34-7365a8a740e3.png)

My 3D models get blurred in iOS when I view them on my website. 
Currently they're uploaded on AWS S3 and model size is around 10MB. 

Can anyone suggest solutions to this issue? 
Thanks.

**Comment by elalish** [comment]
This is due to our dynamic render scaling to maintain frame rate. However, I've heard rumors that iframes on iOS (and maybe other things) could be causing GPU throttling. Please look at our [example](https://modelviewer.dev/examples/loading/#renderScale); it shows how to disable it for instance. Please let us know if it seems that your model seems to be taxing the GPU heavily (drops frame rate when forced to full scale) or not (in which case something fishy might be going on in iOS).

**Reply by nemesis-ren** [reply]
Where do I put the MinimumRenderScale = 1 in my code?
I tried adding it to the <script> but apparently it didn't work.

My code looks like this :

<script type="module" src="https://unpkg.com/@google/model-viewer/dist/model-viewer.min.js"></script>
<script nomodule src="https://unpkg.com/@google/model-viewer/dist/model-viewer-legacy.js"></script>
 
<model-viewer id="roots9menu" 
shadow-intensity="1" 
src="https://roots9menu.s3.ap-south-1.amazonaws.com/GLB/Salad.glb"
ios-src="https://roots9menu.s3.ap-south-1.amazonaws.com/USDZ/Salad.usdz"
alt="Salad" 

auto-rotate 
camera-controls 
min-camera-orbit="auto 45deg auto" 
max-camera-orbit="auto 90deg auto"

ar ar-modes="webxr scene-viewer quick-look">

</model-viewer>
<style>
    #roots9menu {
        width: 300px;
        height: 300px;
    }

**Reply by elalish** [reply]
I can't see where you tried to add it to your script. Have you looked at the example I linked? Keep in mind that the order in which the scripts run is important for static properties.

**Reply by arash-ahmadi** [reply]
> This is due to our dynamic render scaling to maintain frame rate. However, I've heard rumors that iframes on iOS (and maybe other things) could be causing GPU throttling. Please look at our [example](https://modelviewer.dev/examples/loading/#renderScale); it shows how to disable it for instance. Please let us know if it seems that your model seems to be taxing the GPU heavily (drops frame rate when forced to full scale) or not (in which case something fishy might be going on in iOS).

I seem to be facing Dynamic Scaling making objects blurry in AR mode on iOS devices. In the example shown, it is not clear how to disable the dynamic scaling or set a minimum render-scale. Is it possible to set minimum render-scale on AR?

**Reply by elalish** [reply]
AR on iOS happens through Apple's native Quick Look app, which we don't control (we only launch it). Can you show a screen recording of the problem?
